protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -155,4 +163,28 @@ protected Closeable addContextToMDC(AuthenticateRequest request) {
             .addToContext("authType", request.getAuthType())
             .build();
     }
+
+    private void doOAuth(String jwtToken, ImapSession session, ImapRequest request, Responder responder) {
+        Preconditions.checkArgument(session.oidcSaslConfiguration().isPresent(), "Can not do SASL when the oidcSasl configuration is not yet");
+        OidcSASLConfiguration configuration = session.oidcSaslConfiguration().get();
+        Optional<String> claim = new OidcJwtTokenVerifier().verifyAndExtractClaim(jwtToken, configuration.getJwksURL(), configuration.getClaim());
+
+        if (claim.isPresent()) {
+            final MailboxManager mailboxManager = getMailboxManager();
+            try {
+                Username username = Username.fromMailAddress(new MailAddress(claim.get()));
+                MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
+                session.authenticated();
+                session.setMailboxSession(mailboxSession);
+                okComplete(request, responder);
+            } catch (AddressException e) {
+                LOGGER.error("Error encountered while login. mail address is invalid, {}", claim.get(), e);
```

Comment:
How can we handler "user not found" case? 
Add UserRepository in this layer is will accepted?

Reply:
I see in https://github.com/apache/james-project/pull/784/commits/f0806a65a1e4f47be076afae3c3fd16f6e79bc78, if the jwt is verified we just then create the session directly.

Not really sure about this too, hope to get another ans.

Reply:
I agree adding a check to the users repository.

An alternative could be "auto-provisionning" : creating the user on its first interaction, but it seems overkill to me...

Reply:
I tried to embed UserRepository in here, but it needs many boilerplate, and is complex to another class that invokes AuthenticateProcessor.java.